docs/source-2.0/aws/aws-endpoints.rst

```diff
@@ -0,0 +1,375 @@
+    service MyService {
+        version: "2020-04-02"
+    }
+
+While services are strongly encouraged to follow standard endpoint patterns,
+there are occasional exceptions and special cases.  The following example defines
+a service that use standard regional endpoints, but uses a non-standard pattern for
+FIPS endpoints in US GovCloud:
+
+.. code-block:: smithy
+
+    @standardRegionalEndpoints{
+        partitionSpecialCases: [
+            {
+                partition: "aws-us-gov",
+                endpoint: "myservice.{region}.{dnsSuffix}",
+                fips: true
+            }
+        ]
+    }
+    service MyService {
+        version: "2020-04-02"
+    }
+
+---------------------------
+PartitionSpecialCase object
+---------------------------
+
+A PartitionSpecialCase object contains the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property name
+      - Type
+      - Description
+    * - partition
+      - ``string``
+      - **Required**. The partition to special case (example: "aws").
+    * - endpoint
+      - ``string``
+      - **Required**. The special cased endpoint template.
+    * - dualStack
+      - ``boolean``
+      - When ``true`` the special case will apply to dualstack endpoint variants.
+    * - fips
+      - ``boolean``
+      - When ``true`` the special case will apply to fips endpoint variants.
+
+------------------------
+RegionSpecialCase object
+------------------------
+
+A RegionSpecialCase object contains the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property name
+      - Type
+      - Description
+    * - region
+      - ``string``
+      - **Required**. The region to special case (example: "us-west-2").
+    * - endpoint
+      - ``string``
+      - **Required**. The special cased endpoint template.
+    * - dualStack
+      - ``boolean``
+      - When ``true`` the special case will apply to dualstack endpoint variants.
+    * - fips
+      - ``boolean``
+      - When ``true`` the special case will apply to fips endpoint variants.
+    * - signingRegion
+      - ``string``
+      - Override the signingRegion used for this region.
+
+.. smithy-trait:: aws.endpoints#nonRegionalizedEndpoints
+.. _aws.endpoints#nonRegionalizedEndpoints-trait:
+
+-------------------------------------------------
+``aws.endpoints#nonRegionalizedEndpoints`` trait
+-------------------------------------------------
+
+Summary
+    Indicates that a service is non-regionalized and a single endpoint should be resolved
+    per partition.
+
+Trait selector
+    ``service``
+Trait value
+    A ``structure`` with the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property
+      - Type
+      - Description
+    * - endpointPattern
+      - ``string``
+      - **Required** The pattern to use for the partition endpoint.  This value can be set to ``service_dnsSuffix`` to
```

Comment:
Are these values special cased and this can be anything, but these are shortcuts? Or are they they only valid values?

Reply:
This is intended to be an enum and those are the only two valid values.  Is there a prefered way to document it?  I was referencing `httpApiKeyAuth` trait's "in" member, which just has "This value can be set to header or query."

Reply:
I think the wording just threw me off - It's a type not the explicit pattern. "The pattern type to use for the partition endpoint", since it's not the literal pattern to use.